### DISCUSSION TITLE
Z Full Step setting

### DISCUSSION THREAD

**Discussion by NameOfTheDragon** [supermerill/SuperSlicer#1149] [Q&A]
How is the `Z Full Step` setting used? I'm confused about what effect it has and some of the values I'm seeing in there.

The term "Full Step" typically implies that microsteps are ignored so for a 1.8 degree stepper, there would be 200 full steps per revolution. However, some values I'm seeing in there are too small and appear to incorporate microsteps. Let's consider some examples.

- For an Ender 3 with 8mm pitch lead screw, you'd expect to see `8 / 200 full steps = 0.04 mm`. And that's the value in the default Ender 3 profile, so all good.
-   For a Voron 2, you have a 20 tooth drive pulley which gives 40mm per revolution, that's geared by 80:16, so we'd expect `40/80*16/200 = 0.04 mm`. But the SS Voron profile comes with this value set to 0.00625. If we include microsteps in the calculation, `0.04/16 = 0.0025 mm`. If I'm doing my calculation correctly (which may not be the case) then the built in default seems wrong in each case.

So how is this value used and what's the correct setting for a Voron V2?

**Comment by supermerill** [comment]
it ensures that every layer height is a multiple of this value.
It's useful if you don't want the z motor to microstep and use one of the variable layer height feature.

note that it's almost useless if your z=0 is not at a full step. So some gantry levelling options may put your z motor(s) already on a microstep. So for a voron2, I don't think it's useful because of the quad_gantry_levelling, that's why I put it on a microstep.
